### PR TITLE
Add mstest runner samples

### PR DESCRIPTION
**Pull request by nohwnd**
Add simple example and docker example.

### CODE REVIEW

File: samples/mstest-runner/RunInDocker/MyServer.Tests/GlobalUsings.cs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+global using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (Evangelink) [inline samples/mstest-runner/RunInDocker/MyServer.Tests/GlobalUsings.cs]: It's more common to use the using in csproj nowadays.

File: samples/mstest-runner/RunInDocker/MyServer.Tests/MyServer.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <PropertyGroup>
+    <EnableMSTestRunner>true</EnableMSTestRunner>
+    <OutputType>Exe</OutputType>
+    
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
```
Comment (Evangelink) [inline samples/mstest-runner/RunInDocker/MyServer.Tests/MyServer.Tests.csproj]: This isn't needed

File: samples/mstest-runner/Simple1/TestConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1 @@
+[assembly: Parallelize]
```
Comment (Evangelink) [inline samples/mstest-runner/Simple1/TestConfiguration.cs]: It's better to use the "best parallelization mode".

```suggestion
[assembly: Parallelize(Scope = ExecutionScope.MethodLevel)]
```

File: samples/mstest-runner/Simple1/Simple1.csproj [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    <!-- Enable the MSTest runner, this is an opt-in feature -->
+    <EnableMSTestRunner>true</EnableMSTestRunner>
+    <OutputType>Exe</OutputType>
+
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
```
Comment (Evangelink) [inline samples/mstest-runner/Simple1/Simple1.csproj]: Un-needed

```suggestion
```

**Review by Evangelink — DISMISSED** [review]

File: samples/mstest-runner/RunInDocker/MyServer.Tests/MyServer.Tests.csproj (hunk 2 above)
Comment (nohwnd) [reply samples/mstest-runner/RunInDocker/MyServer.Tests/MyServer.Tests.csproj]:
```suggestion
```

**Pull request merged**